### ISSUE TITLE
[apex] Avoid debug statements since it impact performance

### ISSUE THREAD

**Issue by jonathanwiesel** [pmd/pmd#3307]
**Proposed Rule Name:** AvoidDebugStatements

**Proposed Category:** Performance

**Description:**

As demonstrated by @keirbowden in [his blog](http://bobbuzzard.blogspot.com/2021/05/the-impact-of-systemdebug.html), the usage of debug statements generates a negative impact in the application performance even when debug traces are not set.

Taking into account that this technique is usually implemented temporarily and should not reach production environments it seems important to flag their usage to mitigate performance issues (especially taking into consideration that most of what is achieved with a `System.debug` statement can also be done using the Apex Replay Debugger and checkpoints).

Obviously there are exception where it may be needed and for those use cases the PMD warning suppression mechanism could be used to explicitly indicate that is actually needed (like the `AvoidGlobalModifier` one).

**Code Sample:** 

Should be flagged

```
public class Foo {
  public void debugMe() {
    System.debug(LoggingLevel.ERROR, 'please flag me');
  }
}
```

Good to go

```
@SuppressWarnings('PMD.AvoidDebugStatements')
public class LoggingManager {
  public static void log(System.LoggingLevel level, Object data) {
    System.debug(level, data);
  }
}
```

```
public class Foo {
  public void debugMe() {
    LoggingManager.log(LoggingLevel.ERROR, 'wont flag');
  }
}
```

**Comment by rsoesemann** [comment]
@jonathanwiesel Please see my comment and doubt on you PR.
